docs/api/expect.md

```diff
@@ -1329,6 +1329,140 @@ test('spy function returns bananas on second call', async () => {
 })
 ```
 
+## Chai-Style Spy Assertions
```

Comment:
This is a guide, not an API.

For an API we need all methods to be described similar to other `to*` assertions

The current doc can be moved to guides